Compilation errors in pif_solaris_ipv6.c

<!--Thank you for taking the time to submit an issue!-->

## Background information
I am trying to compile a cloned version of OpenMPI including OpenPMIx under Solaris 11.4 where I get compilation errors.
### What version of the PMIx Reference Library are you using? <!--(e.g., v1.0, v2.1, git master @ hash, etc.)-->
Commit https://github.com/openpmix/openpmix/commit/5bd97056ccd994c711fde4ac19aded8e4eac8285
### Describe how PMIx was installed <!--(e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)-->
OpenMPI cloned from current release, i.e. commit dd6b87590cd767f5ff2047ce117a4f16a216b628, followed by

```sh
export FC=/opt/ulm/dublin/cmd/gfortran
export CC=/opt/ulm/dublin/cmd/gcc
export CPPFLAGS=-D_XOPEN_SOURCE=700
export CXX=/opt/ulm/dublin/cmd/g++
export AUTOMAKE_JOBS=8

cd ompi-master &&
   perl autogen.pl &&
   bash ./configure --prefix=/opt/ulm/dublin \
      --disable-silent-rules \
      --libdir=/opt/ulm/dublin/lib/amd64 \
      --enable-wrapper-rpath &&
   make DESTDIR=/home/pkgdev/dublin/openmpi/proto install
```

### Please describe the system on which you are running

* Operating system/version: Solaris 11.4
* Compiler: gcc 11.2.0
* Computer hardware: 64 bit x86 architecture
* Network type: not relevant here

-----------------------------

## Details of the problem

Compilation errors for pif_solaris_ipv6.c:

```
libtool: compile:  /opt/ulm/dublin/cmd/gcc -DHAVE_CONFIG_H -I. -I../../../../src/include -I../../../../include -iquote/home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/openpmix -iquote/home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/openpmix/src -I/home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/openpmix/include -I/home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/hwloc-2.7.1/include -I/home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/hwloc-2.7.1/include -D_REENTRANT -O3 -DNDEBUG -D_REENTRANT -finline-functions -Wundef -Wno-long-long -Wsign-compare -Wmissing-prototypes -Wstrict-prototypes -Wcomment -Wshadow -Werror-implicit-function-declaration -fno-strict-aliasing -pedantic -Wall -Wextra -Werror -MT pif_solaris_ipv6.lo -MD -MP -MF .deps/pif_solaris_ipv6.Tpo -c pif_solaris_ipv6.c  -fPIC -DPIC -o .libs/pif_solaris_ipv6.o
pif_solaris_ipv6.c: In function 'if_solaris_ipv6_open':
pif_solaris_ipv6.c:136:43: error: comparison of integer expressions of different signedness: 'long unsigned int' and 'int' [-Werror=sign-compare] 
  136 |     for (i = 0; i + sizeof(struct lifreq) <= lifconf.lifc_len; i += sizeof(*lifreq)) {
      |                                           ^~
pif_solaris_ipv6.c:173:70: error: format '%d' expects argument of type 'int', but argument 3 has type 'long unsigned int' [-Werror=format=]
  173 |                     pmix_output(0, "pmix_ifinit: unable to allocate %d bytes\n",
      |                                                                     ~^
      |                                                                      |
      |                                                                      in
      |                                                                     %ld
  174 |                                 sizeof(pmix_pif_t));
      |                                 ~~~~~~~~~~~~~~~~~~
      |                                 |
      |                                 long unsigned int
pif_solaris_ipv6.c:95:14: error: variable 'kindex' set but not used [-Werror=unused-but-set-variable]
   95 |     uint16_t kindex;
      |              ^~~~~~
cc1: all warnings being treated as errors
```

Comments:

- _i_ is of type int, adding size_t (which is unsigned) to int yields size_t, `lifconf.lifc_len` is of type int, see https://docs.oracle.com/cd/E88353_01/html/E37851/if-tcp-4p.html. I suggest to convert the latter to size_t.
- size_t values (type returned by sizeof) should be printed using the format "%zu", not "%d". This is a real error.
- The warning regarding _kindex_ is correct, the variable gets assigned to in line 147 but is nowhere being used.

--- a/src/mca/pif/solaris_ipv6/pif_solaris_ipv6.c
+++ b/src/mca/pif/solaris_ipv6/pif_solaris_ipv6.c
@@ -4,7 +4,7 @@
  * Copyright (c) 2016      Research Organization for Information Science
  *                         and Technology (RIST). All rights reserved.
  * Copyright (c) 2016-2020 Intel, Inc.  All rights reserved.
- * Copyright (c) 2021-2022 Nanook Consulting.  All rights reserved.
+ * Copyright (c) 2021-2023 Nanook Consulting.  All rights reserved.
  * $COPYRIGHT$
  *
  * Additional copyrights may follow
@@ -92,7 +92,6 @@ static int if_solaris_ipv6_open(void)
     int i;
     int sd;
     int error;
-    uint16_t kindex;
     struct lifnum lifnum;
     struct lifconf lifconf;
     struct lifreq *lifreq, lifquery;
@@ -133,7 +132,7 @@ static int if_solaris_ipv6_open(void)
         pmix_output(0, "pmix_ifinit: IPv6 SIOCGLIFCONF failed with errno=%d\n", errno);
     }
 
-    for (i = 0; i + sizeof(struct lifreq) <= lifconf.lifc_len; i += sizeof(*lifreq)) {
+    for (i = 0; i + sizeof(struct lifreq) <= (size_t)lifconf.lifc_len; i += sizeof(*lifreq)) {
 
         lifreq = (struct lifreq *) ((caddr_t) lifconf.lifc_buf + i);
         pmix_strncpy(lifquery.lifr_name, lifreq->lifr_name, sizeof(lifquery.lifr_name) - 1);
@@ -144,7 +143,6 @@ static int if_solaris_ipv6_open(void)
             pmix_output(0, "pmix_ifinit: SIOCGLIFINDEX failed with errno=%d\n", errno);
             return PMIX_ERROR;
         }
-        kindex = lifquery.lifr_index;
 
         /* lookup interface flags */
         error = ioctl(sd, SIOCGLIFFLAGS, &lifquery);
@@ -170,7 +168,7 @@ static int if_solaris_ipv6_open(void)
 
                 intf = PMIX_NEW(pmix_pif_t);
                 if (NULL == intf) {
-                    pmix_output(0, "pmix_ifinit: unable to allocate %d bytes\n",
+                    pmix_output(0, "pmix_ifinit: unable to allocate %zu bytes\n",
                                 sizeof(pmix_pif_t));
                     return PMIX_ERR_OUT_OF_RESOURCE;
                 }

